### ISSUE TITLE
SqlCeToolbox 3.5 4.7.601.0 crashes on listing tables of a database

### ISSUE THREAD

**Issue by PeterAdam** [ErikEJ/SqlCeToolbox#750]
I have registered an SqlCe 3.5 file, opened it in the treeview but when tried to open the Tables node the program crashed.

Build 4.7.574 is working with the same file. File is 3.5 version, checked with the Toolbox.

### Further technical details
Toolbox/Power Tools version: 4.7.601.0

Database engine: SQL Compact 3.5.8080.0

Visual Studio or SSMS version: standalone

**Comment by icnocop** [comment]
Are there any relevant entries in the windows event log?

**Comment by PeterAdam** [comment]
A hibát okozó alkalmazás neve: SqlCe35Toolbox.exe, verzió: 4.7.601.0, időbélyeg: 0x5bbe4bbe
A hibát okozó modul neve: KERNELBASE.dll, verzió: 6.1.7601.24168, időbélyeg: 0x5b1aa77b
Kivételkód: 0xe0434352
Hiba pozíciója: 0x0000845d
A hibát okozó folyamat azonosítója: 0x1e38
A hibát okozó alkalmazás indításának időpontja: 0x01d465dbb22f6b6a
A hibát okozó alkalmazás elérési útja: C:\Users\DBA\Downloads\SqlCe35Toolbox.exe
A hibát okozó modul elérési útja: C:\Windows\system32\KERNELBASE.dll
Jelentés azonosítója: 66e5e7ee-d1cf-11e8-9978-38d5472687c8

**Comment by PeterAdam** [comment]
Alkalmazás: SqlCe35Toolbox.exe
Keretrendszer verziószáma: v4.0.30319
Leírás: A folyamat egy nem kezelt kivétel miatt meg lett szakítva.
Kivétel adatai: System.IndexOutOfRangeException
   a következő helyen: System.Data.SqlServerCe.SqlCeDataReader.IsDBNull(Int32)
   a következő helyen: System.Data.SqlServerCe.SqlCeDataReader.GetBoolean(Int32)
   a következő helyen: ErikEJ.SqlCeScripting.DBRepository.AddToListColumns(System.Collections.Generic.List`1<ErikEJ.SqlCeScripting.Column> ByRef, System.Data.SqlServerCe.SqlCeDataReader)
   a következő helyen: ErikEJ.SqlCeScripting.DBRepository.ExecuteReader[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]](System.String, AddToListDelegate`1<System.__Canon>)
   a következő helyen: ErikEJ.SqlCeScripting.DBRepository.GetAllColumns()
   a következő helyen: ErikEJ.SqlCeToolbox.ToolWindows.ExplorerControl.FillTableItems(System.Collections.Generic.KeyValuePair`2<System.String,System.String>, ErikEJ.SqlCeToolbox.Helpers.DatabaseTreeViewItem, System.Collections.Generic.IList`1<System.String>)

Kivétel adatai: System.Reflection.TargetInvocationException
   a következő helyen: System.RuntimeMethodHandle.InvokeMethod(System.Object, System.Object[], System.Signature, Boolean)
   a következő helyen: System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(System.Object, System.Object[], System.Object[])
   a következő helyen: System.Delegate.DynamicInvokeImpl(System.Object[])
   a következő helyen: System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   a következő helyen: System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   a következő helyen: System.Windows.Threading.DispatcherOperation.InvokeImpl()
   a következő helyen: System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   a következő helyen: System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   a következő helyen: System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   a következő helyen: System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   a következő helyen: MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   a következő helyen: System.Windows.Threading.DispatcherOperation.Invoke()
   a következő helyen: System.Windows.Threading.Dispatcher.ProcessQueue()
   a következő helyen: System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   a következő helyen: MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   a következő helyen: MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   a következő helyen: System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   a következő helyen: System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   a következő helyen: System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   a következő helyen: MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
   a következő helyen: MS.Win32.UnsafeNativeMethods.DispatchMessage(System.Windows.Interop.MSG ByRef)
   a következő helyen: System.Windows.Threading.Dispatcher.PushFrameImpl(System.Windows.Threading.DispatcherFrame)
   a következő helyen: System.Windows.Threading.Dispatcher.PushFrame(System.Windows.Threading.DispatcherFrame)
   a következő helyen: System.Windows.Application.RunDispatcher(System.Object)
   a következő helyen: System.Windows.Application.RunInternal(System.Windows.Window)
   a következő helyen: System.Windows.Application.Run(System.Windows.Window)
   a következő helyen: SqlCeToolboxExe.App.Main()

**Comment by ErikEJ** [comment]
@icnocop  Looks like it is related to your latest fix?

**Comment by icnocop** [comment]
@ErikEJ , yes. It has a similar stack trace as reported in https://github.com/ErikEJ/SqlCeToolbox/commit/5908ca5bd5410c74bd960bef6d94bc10fdeffd96#commitcomment-30845571. I'll take a look. Thank you.

**Comment by ErikEJ** [comment]
Yes, the tests are in not so good state, sorry.

**Comment by icnocop** [comment]
I created PR #751 but I couldn't reproduce the issue in the unit test.

I was able to reproduce the issue using SqlCe35Toolbox.exe.

It seems the project `SqlCe35ToolboxExe .csproj` references `SqlCeScripting.dll` from `.\src\GUI\lib` instead of `.\src\API\bin\Debug` or `.\src\API\bin\Release` and this is why SqlCe32Toolbox.exe is crashing; it's referencing an older version of `SqlCeScripting.dll` which doesn't have the fix.

**Comment by ErikEJ** [comment]
Thanks @icnocop 

@PeterAdam Are you using the VS extension or the standalone tool?
